src/rgw/rgw_data_sync.cc

```diff
@@ -855,7 +859,14 @@ class RGWReadRemoteBucketIndexLogInfoCR : public RGWCoroutine {
 	                                { NULL, NULL } };
 
         string p = "/admin/log/";
-        call(new RGWReadRESTResourceCR<rgw_bucket_index_marker_info>(sync_env->cct, sc->conn, sync_env->http_manager, p, pairs, info));
+
+        if (error_inject &&
+            rand() % 10000 < cct->_conf->rgw_read_bilog_info_inject_err_probability * 10000.0) {
+          tn->log(0, SSTR("injecting read bilog info error on key=" << instance_key));
```

Comment:
don't forget to move this from `RGWReadRemoteBucketIndexLogInfoCR` into `RGWDataFullSyncSingleEntryCR`

when i set `rgw_read_bilog_info_inject_err_probability=1` and ran `test_resharded_bucket_full_sync_after_data_sync_init`, i do see `RGWDataFullSyncSingleEntryCR` write its error repo with empty gen:
```
2022-07-22T16:31:25.562-0400 7fd81e7fc640  0 RGW-SYNC:data:sync:shard[0]: injecting read bilog info error on key=ruxtxm-1:22d1b0f0-e727-44b8-83a0-86f6a4c0b10e.4146.1
2022-07-22T16:31:25.562-0400 7fd81e7fc640 20 rgw rados thread: cr:s=0x55b85a4a0df0:op=0x55b85a30c6b0:33RGWReadRemoteBucketIndexLogInfoCR: operate()
2022-07-22T16:31:25.562-0400 7fd81e7fc640 20 rgw rados thread: cr:s=0x55b85a4a0df0:op=0x55b85a30c6b0:33RGWReadRemoteBucketIndexLogInfoCR: operate() returned r=-2
2022-07-22T16:31:25.562-0400 7fd81e7fc640 20 rgw rados thread: cr:s=0x55b85a4a0df0:op=0x55b85a30d8a0:28RGWDataFullSyncSingleEntryCR: operate()
2022-07-22T16:31:25.562-0400 7fd81e7fc640 10 RGW-SYNC:data:sync:shard[0]: full sync: failed to read remote bucket info. Writing 0 to error repo for retry
2022-07-22T16:31:25.562-0400 7fd81e7fc640 20 rgw rados thread: cr:s=0x55b85a4a0df0:op=0x55b85a30c6b0:N3rgw10error_repo19RGWErrorRepoWriteCRE: operate()
```
after switching to data incremental sync, it finds that empty gen in the error repo:
```
2022-07-22T16:31:25.567-0400 7fd81e7fc640 20 RGW-SYNC:data:sync:shard[0]: read error repo, got 1 entries
2022-07-22T16:31:25.567-0400 7fd81e7fc640 10 RGW-SYNC:data:sync:shard[0]: gen is --
2022-07-22T16:31:25.567-0400 7fd81e7fc640 20 RGW-SYNC:data:sync:shard[0]: shard_id=0 sync_marker=00000000000000000000:00000000000000022696
2022-07-22T16:31:25.567-0400 7fd81e7fc640 20 rgw rados thread: cr:s=0x55b85a4a0df0:op=0x55b85a30c6b0:38RGWDataIncrementalSyncFullObligationCR: operate()
2022-07-22T16:31:25.567-0400 7fd81e7fc640 20 rgw rados thread: cr:s=0x55b85a31f710:op=0x55b85a30d8a0:27RGWReadRemoteDataLogShardCR: operate()
```
but error injection kicks in again, so it keeps retrying and failing:
```
2022-07-22T16:31:25.568-0400 7fd81e7fc640  0 RGW-SYNC:data:sync:shard[0]:error_repo[ruxtxm-1:22d1b0f0-e727-44b8-83a0-86f6a4c0b10e.4146.1:0]: injecting read bilog info error on key=ruxtxm-1:22d1b0f0-e727-44b8-83a0-86f6a4c0b10e.4146.1
2022-07-22T16:31:25.568-0400 7fd81e7fc640 20 rgw rados thread: cr:s=0x55b85a4a0df0:op=0x55b85a30df20:33RGWReadRemoteBucketIndexLogInfoCR: operate()
2022-07-22T16:31:25.568-0400 7fd81e7fc640 20 rgw rados thread: cr:s=0x55b85a4a0df0:op=0x55b85a30df20:33RGWReadRemoteBucketIndexLogInfoCR: operate() returned r=-2
2022-07-22T16:31:25.568-0400 7fd81e7fc640 20 rgw rados thread: cr:s=0x55b85a4a0df0:op=0x55b85a30c6b0:38RGWDataIncrementalSyncFullObligationCR: operate()
2022-07-22T16:31:25.568-0400 7fd81e7fc640 20 rgw rados thread: cr:s=0x55b85a4a0df0:op=0x55b85a30c6b0:38RGWDataIncrementalSyncFullObligationCR: operate() returned r=-2
```